Dockerfile

```diff
@@ -3,28 +3,33 @@ FROM golang:1.17-bullseye AS builder
 
 WORKDIR /workspace
 
-# Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
+RUN ["go",  "mod", "download"]
 
-COPY hack/ hack/
-COPY helm-plugins/ helm-plugins/
 COPY Makefile.specialresource.mk Makefile.specialresource.mk
 COPY Makefile.helm.mk Makefile.helm.mk
 COPY Makefile.helper.mk Makefile.helper.mk
 COPY Makefile Makefile
+RUN ["make", "controller-gen"]
```

Comment:
Side remark - we probably want to drop this in the future, and rather check in the CI that the code is correctly generated.